### PR TITLE
Feat: integration tests mock handler

### PR DESCRIPTION
**Pull request by foxpy**
This PR introduces ability to mock failing behaviour of tx query callback handler.

### CODE REVIEW

**Review by oopcode — APPROVED** [review]

**Review by ratik — DISMISSED** [review]

File: contracts/neutron_interchain_queries/src/contract.rs [hunk 1]
```diff
@@ -109,6 +121,12 @@ pub fn execute(
             new_update_period,
         } => update_interchain_query(query_id, new_keys, new_update_period),
         ExecuteMsg::RemoveInterchainQuery { query_id } => remove_interchain_query(query_id),
+        ExecuteMsg::IntegrationTestsSetTransfersQueryMock { recipient, amount } => {
+            set_transfers_query_mock(deps, recipient, amount)
+        }
+        ExecuteMsg::IntegrationTestsUnsetTransfersQueryMock {} => unset_transfers_query_mock(deps),
+        ExecuteMsg::IntegrationTestsSetKvQueryMock {} => set_kv_query_mock(deps),
+        ExecuteMsg::IntegrationTestsUnsetKvQueryMock {} => unset_kv_query_mock(deps),
```
Comment (sotnikov-s) [inline contracts/neutron_interchain_queries/src/contract.rs:129]: I don't think we need to emphasise it's relation to integration tests in any way. it's an example document contract, it should have a simple production-like logic. but these integration tests things only overload it in my opinion
I also think that tx query tests don't belong here cause I cover much more tx query cases in my pr. why do we need 1) two different places where we test tx queries; 2) tx query tests in a file named interchain_kv_query.test?
I only wonder if my logic of testing is appropriate: I imitate a failed tx by sending too many coins the sender doesn't have. seems legit, but what do you think?

Comment (foxpy) [reply contracts/neutron_interchain_queries/src/contract.rs:129]: > I don't think we need to emphasise it's relation to integration tests in any way.

I don't agree. I use "integration tests" in naming because these messages are only **useful** for integration tests, they couldn't ever be used as an example of how to write code for Neutron.

> I also think that tx query tests don't belong here

Yep, you are right! Though I don't want to delete this code, I think it tests a useful corner case. Maybe we should just move tx callback logic in our integration tests suite to your branch instead? I really like deleting code, but deleting tests' code is different.

> I imitate a failed tx by sending too many coins the sender doesn't have. seems legit, but what do you think?

That's a different corner case. I really believe we should leave both tests in our code base, in my opinion each of them is useful.

Comment (pr0n00gler) [reply contracts/neutron_interchain_queries/src/contract.rs:129]: I personally agree with @sotnikov-s. These integration tests only handlers seem weird in our **example** contracts. 
And I think _imitating a failed tx by sending too many coins the sender_ is a pretty good way to get an error.

Comment (foxpy) [reply contracts/neutron_interchain_queries/src/contract.rs:129]: but that's not an error raised by callback, imo it is a different kind of error

Comment (sotnikov-s) [reply contracts/neutron_interchain_queries/src/contract.rs:122]: I singularly don't like this approach because when third party guys will come across this example contract, they will see integration tests stuff and think like "aw shit these poor neutron guys couldn't even prepare with a standalone document contract for us — their customers and cryptobrothers". it kind of emphasises that the main reason of this contract is integration tests, not documentation

**Review by sotnikov-s — APPROVED** [review]
LGTM for now, but imo in the future we should find a way to both test sudo errors and have a nice contract that doesn't contain explicit mentions of integration tests. Agreed with Murad on making a task for refactoring the contract in this terms. @foxpy please create a task for it

**Comment by foxpy** [comment]
> please create a task for it

https://p2pvalidator.atlassian.net/browse/LSC-167

File: contracts/neutron_interchain_queries/src/contract.rs (hunk 1 above)
Comment (foxpy) [reply contracts/neutron_interchain_queries/src/contract.rs:122]: This thread is converted to Jira ticket: https://p2pvalidator.atlassian.net/browse/LSC-167

**Review by oopcode — APPROVED** [review]
